source/CargoHold.cpp

```diff
@@ -504,10 +510,6 @@ int CargoHold::Remove(const Outfit *outfit, int amount)
 
 	amount = min(amount, outfits[outfit]);
 	outfits[outfit] -= amount;
-
-	// Remove the entry if there is no instance of this outfit in this cargo hold.
-	if(!outfits[outfit])
-		outfits.erase(outfit);
 	return amount;
```

Comment:
If you can, I'd prefer a unit test asserting that calling remove doesn't change the size of the containers. But I'll settle for a comment stating the reason why `erase` isn't called (e.g. `// Do not invalidate existing iterators by modifying the container.`)